### PR TITLE
Fix 2609 binary userdata lists

### PR DESCRIPTION
**Pull request by Dingyi189-eng**

### CODE REVIEW

**Review by xinzhoudev — COMMENTED** [review]
Motivation
The BinaryExporter in jMonkeyEngine currently fails to correctly deserialize Savable objects that contain multiple UserData lists. When a scene object carries more than one such list, the binary import process misreads the stream, causing subsequent data to be parsed from the wrong offset. This leads to corrupted deserialization results and potential crashes, as reported in issue #2609.
The root cause lies in how BinaryInputCapsule handles the index pointer after reading certain values. The existing logic did not account for cases where multiple consecutive list structures are present, causing the read cursor to fall out of sync with the actual byte stream layout.

Summary
This PR makes two changes to address the issue:
1. BinaryInputCapsule.java — Index correction on NULL_OBJECT / DEFAULT_OBJECT sentinel values
When reading a long value that matches the NULL_OBJECT (-1) or DEFAULT_OBJECT (-2) sentinel constants defined in BinaryOutputCapsule, the index is now rewound by 4 bytes. This compensates for the fact that BinaryOutputCapsule.deflate() encodes these special int-sized markers in a compressed form, so the reader must not advance the full 8-byte stride in such cases.

⚠️ Note: A concern has been raised (by Gemini Code Assist) that this correction may itself introduce stream corruption when a genuine long value of -1L or -2L is written, since deflate() does not apply the same single-byte compression to 8-byte longs. This edge case should be validated before merge.

2. JmeExporterTest.java — Enable list serialization tests
The testLists flag in the exporter consistency test suite is changed from false to true, ensuring that list serialization and deserialization (including multiple UserData lists) is now covered by automated tests, preventing regression.

File: jme3-core/src/plugins/java/com/jme3/export/binary/BinaryInputCapsule.java [hunk 1]
```diff
@@ -959,6 +959,10 @@ protected long readLong(byte[] content) throws IOException {
         }
         bytes = ByteUtils.rightAlignBytes(bytes, 8);
         long value = ByteUtils.convertLongFromBytes(bytes);
+        if (value == BinaryOutputCapsule.NULL_OBJECT
+                || value == BinaryOutputCapsule.DEFAULT_OBJECT) {
+            index -= 4;
+        }
```
Comment (JNightRider) [inline jme3-core/src/plugins/java/com/jme3/export/binary/BinaryInputCapsule.java:965]: I tested these changes without this code snippet; the test runs without issues, so I suspect it's not related to the problem.

File: jme3-core/src/main/java/com/jme3/scene/UserData.java [hunk 2]
```diff
@@ -138,7 +156,16 @@ public static byte getObjectType(Object type) {
     public void write(JmeExporter ex) throws IOException {
         OutputCapsule oc = ex.getCapsule(this);
         oc.write(type, "type", (byte) 0);
-        
+
+        if (uniqueId == null) {
+            synchronized (this) {
+                if (uniqueId == null) {
```
Comment (JNightRider) [inline jme3-core/src/main/java/com/jme3/scene/UserData.java:162]: I don't understand the double validation; isn't one validation enough?

**Review by JNightRider — COMMENTED** [review]
I believe this PR contains the same changes as this [PR - #2610](https://github.com/jMonkeyEngine/jmonkeyengine/pull/2610), only this one slightly modifies the format of the prefix proposed by the previous pull request.

I'm not so sure about the other one, @riccardobl  was there any reason to close that [PR - #2610](https://github.com/jMonkeyEngine/jmonkeyengine/pull/2610)?

**Comment by riccardobl** [comment]
> I'm not so sure about the other one, @riccardobl was there any reason to close that [PR - #2610](https://github.com/jMonkeyEngine/jmonkeyengine/pull/2610)?

It was more complex than what I had imagined and I didn't trust the robots to make changes that were backward compatible in this specific area, so I wanted to resolve this manually sometimes in the future (has low priority for me atm).

This PR also looks AI generated, so not something I would merge, jme's UserData are kinda messed up and have a lot of side effects
